snap/info_test.go

```diff
@@ -380,6 +382,40 @@ func (s *infoSuite) TestReadCurrentInfo(c *C) {
 	c.Assert(errors.As(err, &snap.NotFoundError{}), Equals, true)
 }
 
+func (s *infoSuite) TestReadCurrentComponentInfo(c *C) {
+	const snapYaml = `
+name: sample
+version: 1
+components:
+ comp:
+   type: test`
+
+	const componentYaml = `
+component: sample+comp
+type: test
+`
+
+	info := snaptest.MockSnapCurrent(c, snapYaml, &snap.SideInfo{
+		Revision: snap.R(42),
+	})
+
+	snaptest.MockComponentCurrent(c, componentYaml, info, snap.ComponentSideInfo{
+		Component: naming.NewComponentRef("sample", "comp"),
+		Revision:  snap.R(21),
+	})
+
+	// TODO: fix this annoying import cycle issue
+	container := func(p string) (snap.Container, error) {
+		return snapdir.New(p), nil
```

Comment:
you can use a hook from snapdir into snap I suppose to fix this?

Reply:
Addressed in 1e559b987688f65f6fbb22737292471d3ad61c5b, first timing adding one of those, so might be wrong.